src/style/style.js

```diff
@@ -849,6 +853,42 @@ class Style extends Evented {
         sourceCache.setFeatureState(sourceLayer, featureId, state);
     }
 
+    removeFeatureState(target: { source: string; sourceLayer?: string; id?: string | number; }, key?: string) {
+        this._checkLoaded();
+        const sourceId = target.source;
+        const sourceLayer = target.sourceLayer;
+        const sourceCache = this.sourceCaches[sourceId];
+        const featureId = parseInt(target.id, 10);
+
+        if (sourceCache === undefined) {
+            this.fire(new ErrorEvent(new Error(`The source '${sourceId}' does not exist in the map's style.`)));
+            return;
+        }
+        const sourceType = sourceCache.getSource().type;
+
+        if (sourceType === 'geojson' && sourceLayer) {
+            this.fire(new ErrorEvent(new Error(`GeoJSON sources cannot have a sourceLayer parameter.`)));
```

Comment:
Wouldn't it be better to silently ignore the sourceLayer in this case instead of firing an error?

Reply:
can do 👍originally did this because it would better surface a silent error in case the user meant to use a `vector` source but inadvertently called the `geojson` one